modules/directives/src/main/scala/scala/build/preprocessing/directives/DirectiveUtil.scala

```diff
@@ -62,9 +62,15 @@ object DirectiveUtil {
   extension (deps: List[Positioned[String]]) {
     def asDependencies: Either[BuildException, Seq[Positioned[AnyDependency]]] =
       deps
-        .map {
-          _.map { str =>
-            DependencyParser.parse(str).left.map(new DependencyFormatError(str, _))
+        .map { positionedDep =>
+          positionedDep.map { str =>
+            DependencyParser.parse(str).left.map { error =>
+              new DependencyFormatError(
+                str,
+                error,
+                positionOpt = positionedDep.positions.headOption
```

Comment:
Why is only the head position passed from the `positionedDep.positions` list? The `DependencyFormatError` should be updated to accept a list of `Position`


```suggestion
                positionOpt = positionedDep.positions
```